Source/Assimil8or/Assimil8or.cpp

```diff
@@ -498,6 +498,235 @@ void Assimil8orValidator::sortContentsOfFolder (juce::ValueTree folderVT)
+            setParseState (ParseState::ParsingPresetSection, &presetActions, "global");
+        }}
+    });
+    
+    // Preset Actions
+    presetActions.insert({
+        {Section::ChannelId, [this, getParameterIndex] () {
+            curChannelSection = presetProperties.addChannel (getParameterIndex ());
+            channelProperties.wrap (curChannelSection, ChannelProperties::WrapperType::client, ChannelProperties::EnableCallbacks::no);
+            curActions = &channelActions;
+            sectionName = {"channel"};
+            setParseState (ParseState::ParsingChannelSection, &channelActions, "channel");
+        }},
+        {Parameter::Preset::NameId, [this] () {
+            presetProperties.setName(value, false);
+        }},
+        {Parameter::Preset::Data2asCVId, [this] () {
+            presetProperties.setData2AsCV(value, false);
+        }},
+        {Parameter::Preset::XfadeACVId, [this] () {
+            presetProperties.setXfadeACV(value, false);
+        }},
+        {Parameter::Preset::XfadeAWidthId, [this] () {
+            presetProperties.setXfadeAWidth(value.getDoubleValue(), false);
+        }},
+        {Parameter::Preset::XfadeBCVId, [this] () {
+            presetProperties.setXfadeBCV(value, false);
+        }},
+        {Parameter::Preset::XfadeBWidthId, [this] () {
+            presetProperties.setXfadeBWidth(value.getDoubleValue(), false);
+        }},
+        {Parameter::Preset::XfadeCCVId, [this] () {
+            presetProperties.setXfadeCCV(value, false);
+        }},
+        {Parameter::Preset::XfadeCWidthId, [this] () {
+            presetProperties.setXfadeCWidth(value.getDoubleValue(), false);
+        }},
+        {Parameter::Preset::XfadeDCVId, [this] () {
+            presetProperties.setXfadeDCV(value, false);
+        }},
+        {Parameter::Preset::XfadeDWidthId, [this] () {
+            presetProperties.setXfadeDWidth(value.getDoubleValue(), false);
+        }}
+    });
+    
+    // Channel Actions
+    channelActions.insert({
+        {Section::ZoneId, [this, getParameterIndex]() {
+            // TODO - do we need to check for malformed data, ie more than 8 zones
+            curZoneSection = channelProperties.addZone(getParameterIndex());
+            zoneProperties.wrap(curZoneSection, ZoneProperties::WrapperType::client, ZoneProperties::EnableCallbacks::no);
+            curActions = &zoneActions;
+            sectionName = {"zone"};
+            setParseState(ParseState::ParsingZoneSection, &zoneActions, "zone");
+        }},
+        {Parameter::Channel::AttackId, [this] () {
+            channelProperties.setAttack(value.getDoubleValue(), false);
+        }},
+        {Parameter::Channel::AttackFromCurrentId, [this] () {
+            channelProperties.setAttackFromCurrent(value.getIntValue() == 1, false);
+        }},
+        {Parameter::Channel::AttackModId, [this] () {
+            auto [cvInput, attackModAmount] = ChannelProperties::getCvInputAndValueFromString(value);
```

Comment:
**const**, I won't mark the rest, but you can go find them. and yes, these were declarations, but doing a 'const' pass is something I do repeatedly on my code. and so reviews are a perfect place to catch them all